datalad/local/add_readme.py

```diff
@@ -82,6 +89,23 @@ def __call__(dataset, filename='README.md', existing='skip'):
         # unlock, file could be annexed
         if lexists(filename):
             dataset.unlock(filename)
+        if not lexists(filename):
+            # if it does not exists, make sure it is saved into Git
+            if dataset.status(
+                    '.gitattributes',
+                    result_renderer='disabled',
+                    return_type='list',)[0]['state'] != 'clean':
+                raise RuntimeError(
+                    'Stopping, because the .gitattributes file is modified'
+                    )
+            dataset.repo.set_gitattributes(
+                [('README.md', {'annex.largefiles': 'anything' if annex
+                                                    else 'nothing'})])
+            dataset.save(
+                path='.gitattributes',
+                message="[DATALAD] Configure README to be in {}".format(
+                    "annex" if annex else "Git")
+            )
```

Comment:
might be worth conditioning this block on `isinstance(ds.repo, AnnexRepo)` to not pester git-puritans with all the git-annex goodness (docstring might need to be adjusted to state that the option is in effect only for annex'ed repos, which is due anyways since if repo is a pure GitRepo we will not save to annex even if annex=True)

Reply:
good idea - done.